### DISCUSSION TITLE
Statistics analogues to node.js agent

### DISCUSSION THREAD

**Discussion by mnutt** [nodejs/undici#1254] [Q&A]
In switching from the standard node.js agent to undici, it looks like undici will offer more granular statistics than the node.js agent which is great but I'm initially hoping to replicate what we pull from the node.js agent to send to prometheus: the `length` of `sockets`, `freeSockets`, and `requests`. I wanted to confirm if these were equivalents (or if there even are equivalents):

```js
// node agent
Object.values(agent.sockets)[0] || []).length
// undici
balancedPool[kClients].reduce((total, pool) => total + pool[kClients].filter(c => c[kConnected]).length, 0)

// node agent
Object.values(agent.freeSockets)[0] || []).length
// undici
balancedPool[kClients].reduce((total, pool) => total + pool[kClients].filter(c => c[kConnected] && !c[kNeedDrain]).length, 0)

// node agent
Object.values(agent.requests)[0] || []).length
balancedPool[kClients].reduce((total, pool) => total + pool[kClients].filter(c => c[kRunning]).length, 0)
```

**Comment by mcollina** [comment]
Would you like to send a PR to add those as public properties? You should not really access internal fields.
